File: SP-UI/src/app/models/product.ts
export interface Product {
    id: number
    name: string
    image: string
    price: number
    discount: number
    priceWithDiscount: number
    description: string
    weight: number
    height: number
    width: number
    length: number
    categoryId: number
    categoryName: string
    departmentId: number
    departmentName: string
}

export interface PageProduct {
    id: number
    name: string
    image: string
    price: number
    discount: number
    priceWithDiscount: number
    weight: number
    height: number
    width: number
    length: number
    description: string
    categoryId: number
    categoryName: string
    departmentId: number
    departmentName: string
}

export interface MenuProduct {
    id: number
    name: string
}

export interface ProductCreationDTO {
    name: string
    image: File
    categoryId: number
    departmentId: number
    price: number
    discount: number
    description: string
    weight: number
    height: number
    width: number
    length: number
}

File: SoloRespira/src/main/java/com/SoloRespira/SoloRespira/entities/Product.java
package com.SoloRespira.SoloRespira.Entities;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import java.io.Serializable;
import lombok.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
// @Builder // https://github.com/mapstruct/mapstruct/issues/2115
@Table()
public class Product extends BaseEntity implements Serializable {

    private Double price;
    private Double discount;
    private String description;
    private Double weight;
    private Double height;
    private Double width;
    private Double length;
    private Integer initialStock;
    private Integer soldUnits;
    
    @OneToOne
    private Image image;

    @ManyToOne
    private Category category;

    @ManyToOne
    private Deparment deparment;
}

